blog/2021/02/julia-1.6-highlights.md

```diff
@@ -0,0 +1,389 @@
+
+Providing reliable, portable binaries to packages is a challenge that all
+packaging environments must face, and while Julia's strategy has always been to
+prioritize reliability and reproducibility over all other concerns, in the past
+it has come at a cost.  Our solution to the problems of reliability and
+reproducibility was to more fully isolate installed binaries and cross-compile
+them ourselves using the [`BinaryBuilder.jl`](https://github.com/JuliaPackaging/BinaryBuilder.jl)
+framework.  Libraries built from `BinaryBuilder.jl` are most often used through
+so-called JLL packages which provide a standardized API that Julia packages can
+use to access the provided binaries.  This ease of use and reliability of
+installation resulted in _vastly_ increased load times as compared to the bad
+old days when Julia packages would blindly `dlopen()` libraries and load
+whatever libraries happened to be sitting on the library search path.  To
+illustrate the issue, in Julia 1.4, loading the GTK+3 stack required **7
+seconds** when it used to take around **500ms** on the same machine.  Through
+many months of hard work and careful investigation, we are pleased to report
+that the same stack of libraries now takes less than **200ms** to load when
+using Julia v1.6 on the same machine.
+
+The cause of this slowdown was multi-faceted and spread across many different
+layers of the Julia ecosystem.  Part of the issue was general compiler latency,
+which has been a focus of the compiler team for some time now, as evidenced by
+Jeff's post in this blogpost.  Another major piece though was general overhead
+incurred by having so many small JLL packages providing bindings; there was
+significant overhead in the loading of each package.  In particular, there was
+code inferrence, code generation and datastructure loading that needed to be
+eliminated if the JLL packages were to be lightweight enough to not affect
+overall load times.  In our experiments, we found that one of the largest
+sources of package load times was in the deserialization of backedge
+information, the links from functions in `Base` back to our packages that would
+cause our functions to be recompiled if there was an invalidation effecting
+that `Base` function.  As counter-intuitive as it may seem, simply using
+a large number of functions from `Base` can very quickly balloon the
+precompilation cache files for your package, causing an increase in loading
+time!  While the increase itself is small, (`3-10ms` at the worst) when you are
+loading many dozens of JLL packages, this adds up quickly.
+
+Our work to slim JLL packages down resulted in the creation of a new package,
+[`JLLWrappers.jl`](https://github.com/JuliaPackaging/JLLWrappers.jl).  This
+package provides macros that auto-generate the bindings necessary for a JLL
+package, and do so by using the minimum number of functions and datastructures
+possible.  By limiting the number of backedges and datastructures, as well as
```

Comment:
```suggestion
possible.  By limiting the number of back edges and data structures, as well as
```

Reply:
I think it's ok as-is. `backedge` is used elsewhere in blogposts. `julia datastructure` has less hits on google though 🤷‍♂️

Reply:
I would keep `backedge` since I see the core devs use that spelling all the time.

But I would change datastructure to data structure.

Reply:
Ok. I'll make that change